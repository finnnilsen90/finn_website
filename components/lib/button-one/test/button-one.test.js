const React = require('react');
const ReactTestUtils = require('react-dom/test-utils');
const renderer = require('react-test-renderer')
let { shallow, mount, render } = require('enzyme');

const Button = require('../button-one.js');

let newButton = new Button();

describe('component connection', () => {

    it('should be defined', () => {
        expect(Button).toBeDefined();
    })

})  
    
describe('<Button />', () => {

    const wrapper = render(<Button />);

    it('renders components', () => {
        expect(wrapper).toMatchSnapshot();
    });

})

describe('<Button /> should render', () => {

    const wrapper = shallow(<Button name='Button' href='/test' type='submit' value='Login'/>);

    it('button name and href', () => {
        expect(wrapper).toMatchSnapshot();
    });

})