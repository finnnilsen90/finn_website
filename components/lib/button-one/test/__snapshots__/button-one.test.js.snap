// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> renders components 1`] = `
<div
  class="button-one_Container"
>
  <button
    class="button_style_one"
  />
</div>
`;

exports[`<Button /> should render button name and href 1`] = `
<div
  className="button-one_Container"
>
  <button
    className="button_style_one"
    href="/test"
    type="submit"
    value="Login"
  >
    Button
  </button>
</div>
`;
